www/src/Services/WebApi.js

```diff
@@ -235,6 +266,25 @@ async function setPinMappings(mappings) {
 		});
 }
 
+async function getProfileOptions(setLoading) {
+	setLoading(true);
+
+	try {
+		const response = await axios.get(`${baseUrl}/api/getProfileOptions`);
+		let profileOptions = { ...baseProfileOptions };
+		response.data['alternativePinMappings'].forEach((altButtons, index) => {
+			for (let prop of Object.keys(altButtons))
+				profileOptions['alternativePinMappings'][index][prop].pin = parseInt(
+					response.data['alternativePinMappings'][index][prop]
+				);
+		});
+		return profileOptions;
```

Comment:
To solve the loading spinner issue, set loading to false before the return:
`setLoading(false);`